hxmath/math/MathTypes.hx

```diff
@@ -157,6 +157,188 @@ abstract QuaternionType(h3d.Quat) from h3d.Quat to h3d.Quat
+        );
+    }
+
+    public var m00(get, set):Float;
+    private inline function get_m00():Float return this._00;
+    private inline function set_m00(v:Float):Float return this._00 = v;
+
+    public var m10(get, set):Float;
+    private inline function get_m10():Float return this._10;
+    private inline function set_m10(v:Float):Float return this._10 = v;
+
+    public var m20(get, set):Float;
+    private inline function get_m20():Float return this._20;
+    private inline function set_m20(v:Float):Float return this._20 = v;
+
+    public var m30(get, set):Float;
+    private inline function get_m30():Float return this._30;
+    private inline function set_m30(v:Float):Float return this._30 = v;
+
+    public var m01(get, set):Float;
+    private inline function get_m01():Float return this._01;
+    private inline function set_m01(v:Float):Float return this._01 = v;
+
+    public var m11(get, set):Float;
+    private inline function get_m11():Float return this._11;
+    private inline function set_m11(v:Float):Float return this._11 = v;
+
+    public var m21(get, set):Float;
+    private inline function get_m21():Float return this._21;
+    private inline function set_m21(v:Float):Float return this._21 = v;
+
+    public var m31(get, set):Float;
+    private inline function get_m31():Float return this._31;
+    private inline function set_m31(v:Float):Float return this._31 = v;
+
+    public var m02(get, set):Float;
+    private inline function get_m02():Float return this._02;
+    private inline function set_m02(v:Float):Float return this._02 = v;
+
+    public var m12(get, set):Float;
+    private inline function get_m12():Float return this._12;
+    private inline function set_m12(v:Float):Float return this._12 = v;
+
+    public var m22(get, set):Float;
+    private inline function get_m22():Float return this._22;
+    private inline function set_m22(v:Float):Float return this._22 = v;
+
+    public var m32(get, set):Float;
+    private inline function get_m32():Float return this._32;
+    private inline function set_m32(v:Float):Float return this._32 = v;
+
+    public var m03(get, set):Float;
+    private inline function get_m03():Float return this._03;
+    private inline function set_m03(v:Float):Float return this._03 = v;
+
+    public var m13(get, set):Float;
+    private inline function get_m13():Float return this._13;
+    private inline function set_m13(v:Float):Float return this._13 = v;
+
+    public var m23(get, set):Float;
+    private inline function get_m23():Float return this._23;
+    private inline function set_m23(v:Float):Float return this._23 = v;
+
+    public var m33(get, set):Float;
+    private inline function get_m33():Float return this._33;
+    private inline function set_m33(v:Float):Float return this._33 = v;
+}
+
+abstract QuaternionType(kha.math.Quaternion) from kha.math.Quaternion to kha.math.Quaternion
+{
+
+    public inline function new(s:Float, x:Float, y:Float, z:Float)
```

Comment:
It is probably worth documenting that Kha's Quaternion class has the real portion last (whereas QuaternionType has it as the first parameter).